### PR TITLE
Items List Feature

### PR DESCRIPTION
**Pull request by LucasFab**
Hello,

This Pull Request is based on the add of the Items List to the Pokedex.

I created the repository, providers, use_cases to manage the data. I had to create my own gist to manage the data of the items, if you want to create it yourself tell me and give me the URL and I will change the URL in the repository.

For the front-end part I chose to stay with something quite similar to the list of the pokemon in the pokedex, meaning that you have a list of the items with an image and the name of the item (and the category of it).

Unfortunately, some items on the json data had a correct URL but there was only one white pixel on this URL for the image so some items don't have an image.

Feel free to comment any suggestions to make it better !

### CODE REVIEW

**Comment by scitbiz** [comment]
Hi @LucasFab, thanks for your contribution!
It will be nice if you could attach some screenshots or videos about your new screen.

**Comment by scitbiz** [comment]
@LucasFab 
Because the repository is in my account so that it's better that I'm the one who keeps all the related stuff about the project.
I've copied your gist, so this is the URL: 
`https://gist.githubusercontent.com/scitbiz/69bb2082702ecbdf3d26cd7ba2a2e00d/raw/pokemon_items.json`

**Comment by LucasFab** [comment]
> Hi @LucasFab, thanks for your contribution! It would be nice if you could attach some screenshots or videos about your new screen.

I will add the screenshots

**Comment by LucasFab** [comment]
> @LucasFab Because the repository is in my account so that it's better that I'm the one who keeps all the related stuff about the project. I've copied your gist, so this is the URL. Thank you! `https://gist.githubusercontent.com/scitbiz/69bb2082702ecbdf3d26cd7ba2a2e00d/raw/pokemon_items.json`

Ok I will make the modification, thanks for the answers

**Comment by LucasFab** [comment]
Simulator Screen Shot - iPhone 12 - 2021-11-26 at 17 41 51

Here is a screenshot of the items list

**Comment by Hugovidafe** [comment]
I've checked the pr with my Redmi Note 8 Pro and I've not noticed any problem ^^

image

File: lib/data/source/github/github_datasource.dart [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 
 class GithubDataSource {
   static const String url =
       'https://gist.githubusercontent.com/scitbiz/0bfdd96d3ab9ee20c2e572e47c6834c7/raw/pokemons.json';
-
+  static const String itemsURL =
+      'https://gist.githubusercontent.com/scitbiz/69bb2082702ecbdf3d26cd7ba2a2e00d/raw/pokemon_items.json';
   GithubDataSource(this.networkManager);
 
   final NetworkManager networkManager;
```
Comment (scitbiz) [inline lib/data/source/github/github_datasource.dart:14]: Please add a new line between static variable and the constructor

File: lib/data/source/local/local_datasource.dart [hunk 2]
```diff
@@ -11,19 +12,20 @@ class LocalDataSource {
...
     await Hive.openBox<PokemonStatsHiveModel>(PokemonStatsHiveModel.boxKey);
+    await Hive.openBox<ItemHiveModel>(ItemHiveModel.boxKey);
   }
 
   Future<bool> hasData() async {
     final pokemonBox = Hive.box<PokemonHiveModel>(PokemonHiveModel.boxKey);
-
-    return pokemonBox.length > 0;
+    final itemBox = Hive.box<ItemHiveModel>(ItemHiveModel.boxKey);
+    return pokemonBox.length > 0 && itemBox.length > 0;
```
Comment (scitbiz) [inline lib/data/source/local/local_datasource.dart:28]: Please add a new line before the return statement

File: lib/domain/entities/item.dart [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+
+class Item {
+  const Item(
+      {@required this.name, @required this.image, @required this.category, @required this.effect});
```
Comment (scitbiz) [inline lib/domain/entities/item.dart:7]:
```suggestion
      {@required this.name, @required this.image, @required this.category, @required this.effect,});
```
Please add a  trailing `,` to make the formatter works correctly

File: lib/routes.dart [hunk 4]
```diff
@@ -42,6 +45,8 @@ class AppNavigator {
       case _Paths.typeEffectsScreen:
         return FadeRoute(page: TypeEffectScreen());
 
+      case _Paths.itemsList:
+        return FadeRoute(page: ItemsListScreen());
       case _Paths.home:
```
Comment (scitbiz) [inline lib/routes.dart:50]: Please add a new line to  separate between cases

File: lib/ui/screens/items_list/items_list.dart [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  void initState() {
+    _fabController = AnimationController(
+      vsync: this,
+      duration: animationDuration,
+    );
+
+    /*_fabAnimation = _fabController.curvedTweenAnimation(
+      begin: 0.0,
+      end: 1.0,
+    );*/
```
Comment (scitbiz) [inline lib/ui/screens/items_list/items_list.dart:28]: Please remove all unnecessary comments

File: lib/ui/screens/items_list/widgets/items_grid.dart [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+  }
+
+  void _onScroll() {
+    if (innerController != null && !innerController.hasClients) return;
+
+    final thresholdReached = innerController.position.extentAfter < _endReachedThreshold;
+
+    if (thresholdReached) {
+      // Load more!
+      context.read(itemStateProvider).getItems();
```
Comment (scitbiz) [inline lib/ui/screens/items_list/widgets/items_grid.dart:50]: Either add a load more feature or remove it. The list is currently duplicating everything when you scrolled  to the bottom of the list

**Review by scitbiz — CHANGES_REQUESTED** [review]
@LucasFab Everything looks good except for the load more feature is missing and the list are currently being duplicated whenever you scrolled to the bottom of the list instead. Please reconsider adding it or remove it completely. Thanks for your contribution!

**Comment by LucasFab** [comment]
I did the modification acording to the PR requirements comments

**Comment by scitbiz** [comment]
Hi @LucasFab, your PR has conflict, please take a look and resolve it

**Comment by hungps** [comment]
Finished in #58

**Pull request closed without merging**